js/graviex.js

```diff
@@ -0,0 +1,231 @@
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const id = markets[i]['id'];
+            const symbolParts = markets[i]['name'].split ('/');
+            const baseId = symbolParts[0];
+            const quoteId = symbolParts[1];
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const active = true;
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': active,
+                'info': markets[i],
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const timestamp = this.milliseconds ();
+        const symbol = market['symbol'];
+        // ticker = ticker['ticker'];
+        const last = this.safeFloat (ticker, 'last');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'sell'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': this.safeFloat (ticker, 'change'),
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': this.safeFloat (ticker, 'volume2'),
+            'info': ticker,
+        };
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'].replace ('_', ''),
+        };
+        const response = await this.publicGetTickers (this.extend (request, params));
+        if (Array.isArray (response)) {
+            const firstTicker = this.safeValue (response, 0, {});
+            return this.parseTicker (firstTicker, market);
+        }
+        return this.parseTicker (response, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetTickers (params);
+        const data = response;
+        // let timestamp = data['at'];
+        // let tickers = data['ticker'];
+        const ids = Object.keys (data);
+        const result = {};
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const market = this.markets_by_id[id];
```

Comment:
The `id` of a market is not necessarily available in `this.markets_by_id`, we have to do the safeMarket logic here, see how it's done for example here: https://github.com/ccxt/ccxt/blob/master/js/coinex.js#L324

Also, please, always add a request sample before each request and a response sample after each line like:

```JavaScript
const response = await this.publicGetTickers (params);
//
//    here
//
```

For example: 

- request sample: https://github.com/ccxt/ccxt/blob/master/js/ftx.js#L1409
- response sample: https://github.com/ccxt/ccxt/blob/master/js/ftx.js#L1453

Reply:
Ok. Fixed